From 4b2f1eece72f64533bd560ea5e8dcd37b83dac88 Mon Sep 17 00:00:00 2001
From: Andreas Farre <andreas.farre@gmail.com>
Date: Thu, 19 Feb 2015 15:48:32 +0100
Subject: [PATCH] WIP

---
 depixel.js | 19 +++++++++++--------
 test.js    | 10 +++++++---
 2 files changed, 18 insertions(+), 11 deletions(-)

diff --git a/depixel.js b/depixel.js
index 3b63853..08a7776 100644
--- a/depixel.js
+++ b/depixel.js
@@ -57,14 +57,17 @@ var depixel = (function() {
         }},
         clone : { enumerable : false, value : function clone() {
             return new Vertex(this.x, this.y);
-        }}
+        }},
+        toString : { enumerable : false, value : function toStrign() {
+                return ["(",this.x, ", ", this.y, ")"].join('');
+        }},
     });
 
     function Node(x, y, color, vertices) {
         this.x = x;
         this.y = y;
         this.edges = [];
-        this.vertices = vertices;
+        this.vertices = vertices || [];
         this.color = color;
     };
 
@@ -165,7 +168,7 @@ var depixel = (function() {
     };
 
     Node.prototype.isEdge = function isEdge() {
-        return this.edges.length > 3;
+        return this.edges.length < 4;
     };
 
     function Graph(pixels, x, y) {
@@ -508,7 +511,7 @@ var depixel = (function() {
 
         do {
             switch (heading) {
-            case RIGHT;
+            case RIGHT:
                 right = current.right();
                 if (right) {
                     next = right.up();
@@ -522,7 +525,7 @@ var depixel = (function() {
                     heading = DOWN;
                 }
                 break;
-            case DOWN;
+            case DOWN:
                 down = current.down();
                 if (down) {
                     next = down.right()
@@ -536,7 +539,7 @@ var depixel = (function() {
                     heading = LEFT;
                 }
                 break;
-            case LEFT;
+            case LEFT:
                 left = current.left();
                 if (left) {
                     next = left.down();
@@ -550,7 +553,7 @@ var depixel = (function() {
                     header = UP;
                 }
                 break;
-            case UP;
+            case UP:
                 up = current.up();
                 if (up) {
                     next = up.left();
@@ -567,7 +570,7 @@ var depixel = (function() {
             }
 
             // add every vertex of current node from 'vertex' to the first common vertex in current node and next node
-            for (var i = vertext, ilen = current.vertices.length; i < ilen; ++i) {
+            for (var i = vertex, ilen = current.vertices.length; i < ilen; ++i) {
                 var v = current.vertices[i];
                 var j = next.vertices.indexOf(v);
                 if (j != -1) {
diff --git a/test.js b/test.js
index b364bdb..e827909 100644
--- a/test.js
+++ b/test.js
@@ -143,9 +143,13 @@ var fn = function (g, i) {
     createCanvas(g, 20);
 };
 
-createCanvas(depixel(new Uint8Array(data1), width, height).createSimilarityGraph().linearize(), 20);
-createCanvas(depixel(new Uint8Array(data4), 4, 3).createSimilarityGraph().linearize(), 40);
-generateTemplateData().forEach(fn);
+//createCanvas(depixel(new Uint8Array(data1), width, height).createSimilarityGraph().linearize(), 20);
+var g;
+createCanvas(g = depixel(new Uint8Array(data2), 8, 8).createSimilarityGraph().linearize(), 20);
+var c = g.contour(g.nodes[0][0])
+alert(c);
+//createCanvas(depixel(new Uint8Array(data4), 4, 3).createSimilarityGraph().linearize(), 40);
+//generateTemplateData().forEach(fn);
 
 /*
 document.body.appendChild(drawCanvas(graph, 20));
-- 
2.1.1

